tests/test_downloader_handlers.py

```diff
@@ -118,6 +120,35 @@ def render(self, request):
         return request.requestHeaders.getRawHeaders(b"content-length")[0]
 
 
+class ChunkedResource(resource.Resource):
```

Comment:
I understand this acts as a valid server response using chunked encoding. Is that correct?
Would it be possible to add a case for a wrong chunked-encoded response?